src/main/scala/io/citrine/lolo/trees/classification/ClassificationTrainingNode.scala

```diff
@@ -1,152 +1,99 @@
 package io.citrine.lolo.trees.classification
 
 import io.citrine.lolo.trees.splits.{NoSplit, Split, Splitter}
-import io.citrine.lolo.trees.{InternalModelNode, ModelNode, TrainingLeaf, TrainingNode}
+import io.citrine.lolo.trees.{InternalModelNode, ModelNode, TrainingNode}
 import io.citrine.lolo.{Learner, PredictionResult}
 import io.citrine.random.Random
 
-/**
-  * @param trainingData     that this node sees
-  * @param leafLearner      to train on leaves
-  * @param split            used to split this node's direct descendants
-  * @param deltaImpurity    change in training set impurity between this node vs direct descendants
-  * @param numFeatures      number of features used in selecting splits
-  * @param minLeafInstances minimum training instances per node
-  * @param remainingDepth   to stop growing the node
-  * @param maxDepth         maximum depth of tree
-  * @param numClasses       total number of classes, used as an output dimensionality
-  * @param splitter         used to select splits
-  * @param rng              random number generator for reproducibility
-  */
-class ClassificationTrainingNode(
+case class ClassificationTrainingNode(
```

Comment:
This is one important change (and the corresponding change to `RegressionTrainingNode`). The case class is now defined with its child nodes and the splitting logic is done in a builder that takes an `rng`.